tests/unit/test_taskgraph_api.py

```diff
@@ -0,0 +1,268 @@
+                'points_df_in': 'points_task.points_df_out'
+            }
+        }
+
+        tspec_list = [points_task_spec, distance_task_spec]
+
+        self.tgraph = TaskGraph(tspec_list)
+
+        # Create a temporary directory
+        self._test_dir = tempfile.mkdtemp()
+        os.environ['GQUANT_CACHE_DIR'] = os.path.join(self._test_dir, '.cache')
+
+    def tearDown(self):
+        global DEFAULT_MODULE
+        os.environ['GQUANT_PLUGIN_MODULE'] = DEFAULT_MODULE
+        os.environ['GQUANT_CACHE_DIR'] = Node.cache_dir
+        shutil.rmtree(self._test_dir)
+
+    @ordered
+    def test_viz_graph(self):
+        '''Test taskgraph to networkx graph conversion for graph visualization.
+        '''
+        nx_graph = self.tgraph.viz_graph(show_ports=True)
+        nx_nodes = [
+            'points_task', 'points_task.points_df_out',
+            'distance_by_cudf', 'distance_by_cudf.distance_df'
+        ]
+        nx_edges = [
+            ('points_task', 'points_task.points_df_out'),
+            ('points_task.points_df_out', 'distance_by_cudf'),
+            ('distance_by_cudf', 'distance_by_cudf.distance_df')
+        ]
+        self.assertEqual(list(nx_graph.nodes), nx_nodes)
+        self.assertEqual(list(nx_graph.edges), nx_edges)
+
+    @ordered
+    def test_build(self):
+        '''Test build of a taskgraph and that all inputs and outputs are set
+        for the tasks withink a taskgraph.
+        '''
+        self.tgraph.build()
+
+        points_node = self.tgraph['points_task']
+        distance_node = self.tgraph['distance_by_cudf']
+
+        onode_info = {
+            'to_node': distance_node,
+            'to_port': 'points_df_in',
+            'from_port': 'points_df_out'
+        }
+        self.assertIn(onode_info, points_node.outputs)
+
+        onode_cols = {
+            'points_df_out': {
+                'x': 'float64',
+                'y': 'float64'
+            }
+        }
+
+        self.assertEqual(onode_cols, points_node.output_columns)
+
+        inode_info = {
+            'from_node': points_node,
+            'from_port': 'points_df_out',
+            'to_port': 'points_df_in'
+        }
+        self.assertIn(inode_info, distance_node.inputs)
+
+        inode_in_cols = {
+            'points_df_in': {
+                'x': 'float64',
+                'y': 'float64'
+            }
+        }
+        self.assertEqual(inode_in_cols, distance_node.input_columns)
+
+        inode_out_cols = {
+            'distance_df': {
+                'x': 'float64',
+                'y': 'float64',
+                'distance_cudf': 'float64'
+            }
+        }
+        self.assertEqual(inode_out_cols, distance_node.output_columns)
+
+    @ordered
+    def test_run(self):
+        '''Test that a taskgraph can run successfully.
+        '''
+        outlist = ['distance_by_cudf.distance_df']
+        # (dist_df_w_cudf, ) = self.tgraph.run(outputs=outlist)
+        (_, ) = self.tgraph.run(outputs=outlist)
```

Comment:
assert the result is correct after the run?

Reply:
https://github.com/rapidsai/gQuant/blob/c501a7595e80a89d5eedbfdf4686f104668b474d/tests/unit/test_taskgraph_api.py#L169-L183

Using numpy random seed to get repeatable results and assert the result is correct.